### PR TITLE
Deprecate project id

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Projects now have a default name, and setting any custom name triggers a warning.

## Motivation and Context

Related to #541. This change provides a softer off-ramp for current users with scripts where they provide the name parameter for project initialization. Rather than removing it entirely, this PR provides a warning that this parameter will be removed in 2.0.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [x] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added all related issue and pull request numbers for future reference (if applicable). See example below.

### CODE REVIEW

**Comment by b-butler** [comment]
@cbkerr @Tobias-Dwyer will you have time to review this PR?

**Review by cbkerr — COMMENTED** [review]
Do we also have to update the docstring of `init_project` on line 3045?

File: signac/contrib/project.py
Comment (vyasr) [reply signac/contrib/project.py]: Good catch! This was necessary in an older version of this code but it shouldn't be any more.

**Review by cbkerr — APPROVED** [review]
Check that my changes to `init_project` on lines 3046 and 3055 are okay.

Otherwise it looks good!

**Comment by vyasr** [comment]
> Check that my changes to `init_project` on lines 3046 and 3055 are okay.
> 
> Otherwise it looks good!

Thanks @cbkerr! Your change looks right to me. I'm going to hold off merging this until #643 is resolved just in case we run into any blockers with the deprecation there.

File: signac/contrib/project.py
Comment (bdice) [inline signac/contrib/project.py:2330]:
```suggestion
                "in favor of using the project root directory to identify projects. "
```

File: signac/contrib/project.py
Comment (bdice) [inline signac/contrib/project.py:3047]:
```suggestion
    """Initialize a project.
```

File: signac/contrib/project.py
Comment (bdice) [inline signac/contrib/project.py:3057]: For consistency, mark all optional parameters here and below:
```suggestion
    root : str, optional
```

File: changelog.txt
Comment (bdice) [inline changelog.txt:30]: I think this changelog can be clearer about our intentions:
```suggestion
 - Project names have a default value of ``'project'``. Project names will no longer be used in signac 2.0.
```

I considered the wording "Project names will be removed in signac 2.0." In my recollection, we're not going to delete the key if it already exists, it will just be ignored. If we're deleting the project name key as part of the schema migration, we might consider "will be removed." Otherwise "will no longer be used" feels most appropriate to me.

**Review by bdice — CHANGES_REQUESTED** [review]

**Comment by vyasr** [comment]
Still waiting on #643.

File: changelog.txt
Comment (vyasr) [reply changelog.txt:30]: We are as discussed in https://github.com/glotzerlab/signac/pull/643#discussion_r776499851, so I can use that wording.

**Comment by vyasr** [comment]
The relevant changes from #643 were separated into #677, which has now been merged onto next, so we should be good to go on this PR if you approve @bdice.

**Review by bdice — APPROVED** [review]
LGTM!

**Pull request merged**
